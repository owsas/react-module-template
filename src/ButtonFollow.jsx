import React from 'react';
import PropTypes from 'prop-types';
import IconPlus from 'react-icons/lib/go/plus';

import ButtonInteract from './ButtonInteract';

const ButtonFollow = props => (
  <ButtonInteract
    type="likes"
    text="Seguir"
    objectId={props.objectId}
    interactionKey={props.interactionKey}
    className={props.className}
    textDone="Siguiendo"
    icon={<IconPlus />}
    onLogin={props.onLogin}
  />
);

ButtonFollow.propTypes = {
  onLogin: PropTypes.func.isRequired,
  objectId: PropTypes.string.isRequired,
  interactionKey: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default ButtonFollow;

