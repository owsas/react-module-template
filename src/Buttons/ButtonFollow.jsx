import React from 'react';
import PropTypes from 'prop-types';
import IconDone from 'react-icons/lib/md/check';
import IconNotDone from 'react-icons/lib/md/add';

import ButtonAct from './ButtonAct';

const ButtonFollow = props => (
  <ButtonAct
    verb="followed"
    done={() => <span><IconDone /> Lo sigues</span>}
    notDone={() => <span><IconNotDone /> Seguir</span>}
    {...props}
  />
);

ButtonFollow.defaultProps = ButtonAct.defaultProps;
ButtonFollow.propTypes = {
  actor: PropTypes.string.isRequired,
  object: PropTypes.string,
  onSave: PropTypes.func,
  onDestroy: PropTypes.func,
};

export default ButtonFollow;
