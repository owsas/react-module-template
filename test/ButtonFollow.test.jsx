import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ButtonFollow from '../src/ButtonFollow';

test('should match snapshot', () => {
  const m = shallow(<ButtonFollow
    onLogin={() => {}}
    className="Promo"
    objectId="E5SyTySkx5"
    interactionKey="promoId"
  />);

  expect(toJSON(m)).toMatchSnapshot();
});
