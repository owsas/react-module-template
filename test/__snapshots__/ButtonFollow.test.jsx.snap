// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<ButtonInteract
  className="Promo"
  icon={<GoPlus />}
  interactionKey="promoId"
  objectId="E5SyTySkx5"
  onLogin={[Function]}
  text="Seguir"
  textDone="Siguiendo"
  type="likes"
/>
`;
